paramak/parametric_components/toroidal_field_coil_round_corners.py

```diff
@@ -0,0 +1,369 @@
+            point of the vertical section (cm)
+        thickness: The thickness in the (X,Z) plane of the toroidal
+        field coils (cm)
+        extrusiondistance: The total extruded thickness of the coils
+            when in the y-direction (centered extrusion)
+        coil_count: The number of coils placed in the model
+            (changing azimuth_placement_angle by dividing 360 by the amount
+            given). Defaults to 1
+        with_inner_leg: Boolean to include the inside of the Coils
+            defaults to False
+        file_name_stp: Defults to "ToroidalFieldCoilRectangleRoundCorners.stp"
+        file_name_stl: Defaults to "ToroidalFieldCoilRectangleRoundCorners.stl"
+        material_tag: Defaults to "outter_tf_coil_mat"
+    """
+
+    def __init__(
+        self,
+        lower_inner_coordinates: Tuple[float, float],
+        mid_point_coordinates: Tuple[float, float],
+        thickness: Union[float, int],
+        distance: float,
+        number_of_coils: int = 1,
+        with_inner_leg: Optional[bool] = False,
+        stp_filename: Optional[str] = "ToroidalFieldCoilRectangleRoundCorners.stp",
+        stl_filename: Optional[str] = "ToroidalFieldCoilRectangleRoundCorners.stl",
+        material_tag: Optional[str] = "outter_tf_coil_mat",
+        **kwargs
+    ) -> None:
+
+        super().__init__(
+            distance=distance,
+            stp_filename=stp_filename,
+            stl_filename=stl_filename,
+            material_tag=material_tag,
+            **kwargs
+        )
+
+        self._lower_inner_coordinates = lower_inner_coordinates[0], lower_inner_coordinates[1]
+        self._mid_point_coordinates = mid_point_coordinates[0], mid_point_coordinates[1]
+        self._thickness = thickness
+        self._distance = distance
+        self._number_of_coils = number_of_coils
+        self._with_inner_leg = with_inner_leg
+        self._inner_leg_connection_points = []
+        self._analyse_attributes = [
+            0,
+            0,
+            0,
+            0
+        ]
+
+        if len(lower_inner_coordinates) != 2 or len(
+                mid_point_coordinates) != 2:
+            raise ValueError(
+                "The input tuples are too long or too short, they must be 2 element long")
+
+        if self._lower_inner_coordinates[0] > self._mid_point_coordinates[0]:
+            raise ValueError(
+                "The middle point's x-coordinate must be larger than the lower",
+                "inner point's x-coordinate")
+
+        # Adding hidden attributes for analyse list population
+        # inner base length of the coil
+        self._base_length = self._mid_point_coordinates[0] - \
+            self._lower_inner_coordinates[0]
+        self._analyse_attributes[0] = self._base_length
+
+        # height of the coil
+        self._height = abs(
+            mid_point_coordinates[1] - lower_inner_coordinates[1]) * 2
+        self._analyse_attributes[1] = self._height
```

Comment:
```suggestion
        # Adding hidden attributes for analyse list population
        # inner base length of the coil
        self._base_length = self._mid_point_coordinates[0] - \
            self._lower_inner_coordinates[0]
        self._analyse_attributes[0] = self._base_length

        # height of the coil
        self._height = abs(
            mid_point_coordinates[1] - lower_inner_coordinates[1]) * 2
        self._analyse_attributes[1] = self._height
```

These attributes might get out of date if they are set in the init and then changed later.

Reply:
perhaps they can be moved to being calculated just before they are needed

Reply:
Good point, I can move them into a method like the find_radii and just call them were needed in a similar way of that method?